marklogic-data-hub/src/main/resources/ml-modules/root/marklogic-data-hub-spark-connector/readRows.sjs

```diff
@@ -25,19 +25,32 @@ const results = [endpointState];
 
 const partitionNumber = endpointConstants.partitionNumber;
 const partition = endpointConstants.initializationResponse.partitions[partitionNumber];
+const optimizationlevel = endpointConstants.optimizationlevel;
 
 if (endpointState.batchNumber <= partition.batchCount) {
   // Determine the min/max rowID of the current batch number
   const batch = partitionLib.getPartitionBatch(partition, endpointState.batchNumber);
 
   // Run the parameterized plan, constraining it to the min and max row ID of the current batch
-  op.import(endpointConstants.initializationResponse.parameterizedPlan)
-    .result(null, {
-      "MIN_ROW_ID": batch.min,
-      "MAX_ROW_ID": batch.max
-    })
-    .toArray()
-    .forEach(row => results.push(row));
+  if(optimizationlevel >= 0) {
```

Comment:
op.import returns a plan, so to avoid duplication here, try this:

```
let thePlan = op.import(...);
if (optimization level is set) thePlan = thePlan.prepare(...);
thePlan.result(...).toArray().forEach(...)
```

Reply:
There's still duplication here of the "result" call. You can avoid that by doing what I showed above - call thePlan = thePlan.prepare if there's an optimization level.

Reply:
Done. Sorry about the confusion.!